src/material.cpp

```diff
@@ -353,6 +353,35 @@ Material::~Material()
   model::material_map.erase(id_);
 }
 
+Material & Material::clone()
+{
+  std::unique_ptr<Material> mat = std::make_unique<Material>();
+  mat->index_ = model::materials.size();
+  mat->set_id(C_NONE);
+
+  // set all other parameters to whatever the calling Material has
+  mat->name_ = name_;
+  mat->nuclide_ = nuclide_;
+  mat->element_ = element_;
+  mat->ncrystal_mat_ = ncrystal_mat_;
+  mat->atom_density_ = atom_density_;
+  mat->density_ = density_;
+  mat->density_gpcc_ = density_gpcc_;
+  mat->volume_ = volume_;
+  mat->fissionable_ = fissionable_;
+  mat->depletable_ = depletable_;
+  mat->p0_ = p0_;
+  mat->mat_nuclide_index_ = mat_nuclide_index_;
+  mat->thermal_tables_ = thermal_tables_;
+  mat->temperature_ = temperature_;
+
+  if (ttb_)
+    mat->ttb_ = std::make_unique<Bremsstrahlung>(*ttb_);
+
+  model::materials.push_back(std::move(mat));
+  return *mat;
```

Comment:
This is actually going to dereference a null pointer:

https://stackoverflow.com/questions/36071220/what-happens-to-unique-ptr-after-stdmove

So instead should return *materials.back().

Reply:
Good catch!